netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,243 @@
+      }
+      else {
+        println(s"Don't show this  ${jsonObjectList.head.eventId}")
+        return
+      }
+      val html = InfoFormatter.toInnerHtml(formattedString)
+
+      if (!jsonContent.trim.isEmpty) {
+        SwingUtilities.invokeLater(() => {
+          val editorPane: JEditorPane = new JEditorPane {
+            self =>
+            addFocusListener(new FocusListener {
+              def focusGained(fe: FocusEvent): Unit = {
+                FindDialog.watch(self)
+              }
+
+              def focusLost(fe: FocusEvent): Unit = {
+                if (!fe.isTemporary) FindDialog.dontWatch(self)
+              }
+            })
+            setDragEnabled(false)
+            setEditable(false)
+            setContentType("text/html")
+            setOpaque(false)
+            setText(html)
+            setCaretPosition(0)
+          }
+
+          val scrollPane = new JScrollPane(editorPane)
+          scrollPane.setPreferredSize(new Dimension(500, 400))
+          val panel = new JPanel(new BorderLayout())
+          panel.add(scrollPane, BorderLayout.CENTER)
+          val options: Array[AnyRef] = Array(I18N.gui.get("common.buttons.ok"))
+
+          val result = JOptionPane.showOptionDialog(
+            null, panel, I18N.gui.get("dialog.interface.newsNotificationTitle"),
+            JOptionPane.DEFAULT_OPTION, JOptionPane.PLAIN_MESSAGE, null, options, options(0)
+          )
+
+          // Check if OK was clicked (index 0 in options array) and hide the NotificationBanner
+          if (result == 0) {
+            setVisible(false) // Hide NotificationBanner
+            prefs.putInt("lastSeenEventId", jsonObjectList.head.eventId)
+          }
+        })
+      }
+    } catch {
+      case e: Exception => throw new Exception(s"Error in showJsonInDialog: ${e.getMessage}", e)
+    }
+  }
+
+
+  // Lazily initialize the Markdown parser and renderer (can be reused for multiple calls)
+  private lazy val markdownParser = Parser.builder().build()
+  private lazy val htmlRenderer = HtmlRenderer.builder().build()
+
+  def formatJsonObjectList(jsonObjectList: List[JsonObject]): String = {
+    jsonObjectList.map { obj =>
+      // Convert fullText from Markdown to HTML
+      val fullTextHtml = htmlRenderer.render(markdownParser.parse(obj.fullText))
+
+      // Format title, date, and the converted HTML fullText
+      s"<h3>* ${I18N.gui.get("dialog.interface.update")}: ${obj.date} -- ${obj.title}</h3>" +
+        s"$fullTextHtml"
+    }.mkString("<html>", "", "</html>")
+  }
+
+  def getJsonObjectHead: Option[String] = {
+    // Check if jsonObjectList is null or empty, and fetch the JSON if needed
+    if (jsonObjectList == null) {
+      val jsonContent = fetchJsonFromUrl(JsonUrl) // Use the static URL here
+      jsonObjectList = parseJsonToList(jsonContent)
+    }
+
+    // Return the title of the first item if jsonObjectList is not null and has elements
+    jsonObjectList match {
+      case null | Nil =>
```

Comment:
It will never be `null`, you only need to check the `Nil` case

Reply:
Done